src/components/app/exclusioncones/ExclusionGeneral.tsx

```diff
@@ -0,0 +1,390 @@
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { Button } from "@/components/ui/button";
+import { ArrowUpDown, Edit2, Trash } from "lucide-react";
+import axios from "axios";
+import { toast } from "@/hooks/use-toast";
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogFooter,
+} from "@/components/ui/dialog";
+import { Label } from "@/components/ui/label";
+
+interface ExclusionCone {
+  id: string;
+  mission: string;
+  angle_limit: number;
+  interfering_satellite: string;
+  satellite_id: string;
+  gs_id: number;
+}
+
+type SortableColumn = "mission" | "angle_limit" | "interfering_satellite" | "satellite_id" | "gs_id";
+
+const ExclusionGeneral: React.FC = () => {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [sortConfig, setSortConfig] = useState<
+    { key: SortableColumn; direction: "asc" | "desc" } | null
+  >(null);
+  const [exCones, setExCones] = useState<ExclusionCone[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+  const [editingCone, setEditingCone] = useState<ExclusionCone | null>(null);
+  const [isEditDialogOpen, setIsEditDialogOpen] = useState(false);
+
+  useEffect(() => {
+    const fetchExclusionCones = async () => {
+      setLoading(true);
+      try {
+        const response = await axios.get<ExclusionCone[]>("/api/v1/excones/");
+        const processedData = response.data.map((cone) => ({
+          ...cone,
+          angle_limit: Number(cone.angle_limit),
+          gs_id: Number(cone.gs_id),
+        }));
+        setExCones(processedData);
+      } catch (err) {
+        setError("Failed to fetch exclusion cones.");
+        console.error(err);
+        toast({
+          title: "Error: " + err,
+          description:
+            "There was an error fetching from the API. Please try again.",
+          variant: "destructive",
+          duration: 5000,
+        });
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchExclusionCones();
+  }, []);
+
+  const handleSort = (key: SortableColumn) => {
+    let direction: "asc" | "desc" = "asc";
+    if (sortConfig && sortConfig.key === key && sortConfig.direction === "asc") {
+      direction = "desc";
+    }
+    setSortConfig({ key, direction });
+  };
+
+  const handleEdit = (cone: ExclusionCone) => {
+    setEditingCone({ ...cone });
+    setIsEditDialogOpen(true);
+  };
+
+  const handleSaveEdit = async () => {
+    if (!editingCone) return;
+
+    try {
+      await axios.patch(`/api/v1/excones/${editingCone.id}`, editingCone);
+      setExCones((prev) =>
+        prev.map((cone) => (cone.id === editingCone.id ? editingCone : cone))
+      );
+      toast({
+        title: "Exclusion cone updated successfully.",
+        description: `Exclusion cone for mission "${editingCone.mission}" was updated.`,
```

Comment:
Missing colour-coded fill of the toast; same applies for satellites; see SS;
```suggestion
        description: `Exclusion cone for mission "${editingCone.mission}" was updated.`,
        variant: "success",
        duration: 5000,
```
![Screenshot 2025-03-21 at 11 02 50 PM](https://github.com/user-attachments/assets/30f49b47-574f-41e5-98be-ae88a39e70a6)